### PR TITLE
Initial Fabric support

### PR DESCRIPTION
**Pull request by sevoku**
[Preview this branch](https://cosmos-explorer-preview.azurewebsites.net/pull/1607?feature.someFeatureFlagYouMightNeed=true)

This adds initial support for Fabric as a new platform.
Data Explorer will be loaded into an iframe inside our Fabric extension iframe.
* Adds new Platform.Fabric option
* Conditionally loads a different Fluent Theme and overrides some default styles
* Hides many advanced features not required in fabric such as Notebooks, Scripts, etc.
* Replaces the resource tree with a new Fluent implementation

### CODE REVIEW

File: src/Explorer/Menus/CommandBar/CommandBarComponentAdapter.tsx [hunk 1]
```diff
@@ -81,15 +83,27 @@ export const CommandBar: React.FC<Props> = ({ container }: Props) => {
     );
   }
 
+  const rootStyle =
+    configContext.platform == Platform.Fabric
```
Comment (languy) [inline src/Explorer/Menus/CommandBar/CommandBarComponentAdapter.tsx:87]:
```suggestion
    configContext.platform === Platform.Fabric
```

File: src/Explorer/Tree2/ResourceTree.tsx [hunk 2]
```diff
@@ -90,7 +90,7 @@ export const ResourceTree2: React.FC<ResourceTreeProps> = ({ container }: Resour
           openItems={openItems}
           onOpenChange={handleOpenChange}
           size="small"
-          style={{ height: "100%" }}
+          style={{ height: "100%", width: "290px" }}
```
Comment (languy) [inline src/Explorer/Tree2/ResourceTree.tsx:93]: Adds an additional scrollbar on the right.

File: src/Main.tsx [hunk 3]
```diff
@@ -67,6 +70,10 @@ const App: React.FunctionComponent = () => {
   const shouldShowModal = useQueryCopilot((state) => state.showFeedbackModal);
 
   const config = useConfig();
+  if (config?.platform == Platform.Fabric) {
```
Comment (languy) [inline src/Main.tsx:73]:
```suggestion
  if (config?.platform === Platform.Fabric) {
```

File: src/Platform/Fabric/FabricPlatform.tsx [hunk 4]
```diff
@@ -0,0 +1,5 @@
+import React from "react";
```
Comment (languy) [inline src/Platform/Fabric/FabricPlatform.tsx:1]: You might want to add a comment on why we had to add this empty component.

**Comment by sevoku** [comment]
I think this first iteration is ready to go. @ato9000 @victor-meng could you review?

**Review by languy — APPROVED** [review]

**Review by victor-meng — APPROVED** [review]

**Review by languy — APPROVED** [review]

**Review by languy — APPROVED** [review]

**Review by languy — APPROVED** [review]

**Pull request merged**